False positive using ancestor pseudo-class (`vue-scoped-css/require-selector-used-inside`)

Minimal reproduction: https://play.vuejs.org/#eNp9kTFPwzAQhf+KdRNIVToglhAqAaoEDICA0UvkHG2KY1u+Swiq8t85u2rpUHWz3/ve6Z29hbsQiqFHKKFi7IKtGRfaKVU17aCMrYluNTASa1g8orW+mosjSDU/4uVK/GtRkfEBG1HKdU0XQl6qIqXVNg013vpYqojNjXZTmpFTgsMMWLLuq10VG/JO+uSEBuO70FqMr4Fb70hDuZuVvFr6/DxnjWOPs71u1mi+T+gbGpOm4S0iYRxQw8HjOq5Qtkz28uMFRzkfzM43vRX6jPmO5G2fOu6w+941UvuIy22fuuAjt271ScuR0dF+qVQ0kVPmNcinPJxZ/b/uVXGdc/KiMP0B8reZ3Q==

This results in the following nonsensical ESLint output, highlighting the space between `:has(div)` and `.test`:

```
  6:10  warning  The selector ` ` is unused in the template    vue-scoped-css/require-selector-used-inside
```

--- a/.changeset/fix-require-selector-has-pseudo.md
+++ b/.changeset/fix-require-selector-has-pseudo.md
@@ -1,0 +1,5 @@
+---
+"eslint-plugin-vue-scoped-css": patch
+---
+
+Fix false positives in `vue-scoped-css/require-selector-used-inside` for selectors that start with ignored pseudo-classes such as `:has(...)`.

--- a/lib/rules/require-selector-used-inside.ts
+++ b/lib/rules/require-selector-used-inside.ts
@@ -163,6 +163,19 @@ export default {
             isUniversalSelector(s) ||
             isVueSpecialPseudo(s),
         );
+      // Combinators adjacent to ignored selectors cannot be verified on their own.
+      while (
+        selectorNodes.length > 0 &&
+        isSelectorCombinator(selectorNodes[0])
+      ) {
+        selectorNodes.shift();
+      }
+      while (
+        selectorNodes.length > 0 &&
+        isSelectorCombinator(selectorNodes[selectorNodes.length - 1])
+      ) {
+        selectorNodes.pop();
+      }
 
       for (let index = 0; index < selectorNodes.length; index++) {
         const selectorNode = selectorNodes[index];

--- a/tests/lib/rules/require-selector-used-inside.ts
+++ b/tests/lib/rules/require-selector-used-inside.ts
@@ -170,6 +170,15 @@ tester.run("require-selector-used-inside", rule as any, {
         :not(.a) {}
         </style>
         `,
+    // ancestor pseudo-class
+    `
+        <template>
+            <div class="test">Hello</div>
+        </template>
+        <style scoped>
+        :has(div) .test {}
+        </style>
+        `,
     // transition
     `
         <template>
@@ -607,6 +616,22 @@ tester.run("require-selector-used-inside", rule as any, {
           column: 13,
           endLine: 8,
           endColumn: 24,
+        },
+      ],
+    },
+    {
+      code: `
+            <template>
+                <div class="test">Hello</div>
+            </template>
+            <style scoped>
+            :has(div) .unused {}
+            </style>
+            `,
+      errors: [
+        {
+          messageId: "unused",
+          data: { selector: ".unused" },
         },
       ],
     },

